libs/ledger-live-common/src/families/stellar/specs.ts

```diff
@@ -56,12 +77,147 @@ const stellar: AppSpec<Transaction> = {
         };
       },
       test: ({ account, accountBeforeTransaction, operation, transaction }) => {
-        expect(account.balance.toString()).toBe(
-          accountBeforeTransaction.balance.minus(operation.value).toString()
+        // We don't know what the final fee will be until after the tx is
+        // submitted. Using higher max fee to make sure tx doesn't time out.
+        expect(account.balance.toNumber()).toBeLessThanOrEqual(
+          accountBeforeTransaction.balance.minus(operation.value).toNumber()
+        );
+
+        if (transaction.memoValue) {
+          expect(operation.extra).toMatchObject({
+            memo: transaction.memoValue,
+          });
+        }
+
+        // Transaction type "payment" can be either "payment" or "create_account"
+        // (if account is not created).
+        const getType = () => {
+          switch (transaction.operationType) {
+            case "payment":
+              return /payment|create_account/;
+            case "changeTrust":
+              return /change_trust/;
+            default:
+              return "";
+          }
+        };
+
+        expect(transaction.operationType).toMatch(getType());
+      },
+    },
+    {
+      name: "add USDC asset",
+      maxRun: 1,
+      transaction: ({ account, bridge, maxSpendable }) => {
+        invariant(maxSpendable.gt(reserve), "XLM balance is too low");
+
+        const assetUSDC = findAssetUSDC<TokenCurrency>(
+          listTokensForCryptoCurrency(account.currency)
+        );
+        invariant(assetUSDC, "USDC asset not found");
+
+        const transaction = bridge.createTransaction(account);
+
+        const updates: Array<Partial<Transaction>> = [
+          {
+            operationType: "changeTrust",
+            // Setting higher max fee here to make sure transaction doesn't
+            // time out.
+            fees: new BigNumber(MAX_FEE),
+          },
+          {
+            assetCode: USDC_CODE,
+            assetIssuer: USDC_ISSUER,
+          },
+        ];
+
+        return {
+          transaction,
+          updates,
+        };
+      },
+      test: ({ account }) => {
+        const assetId = `${USDC_CODE}:${USDC_ISSUER}`;
+        const hasAsset = account.subAccounts?.find((a) =>
+          a.id.endsWith(assetId)
+        );
+        expect(hasAsset).toBeTruthy();
+      },
+    },
+    {
+      name: "move ~50% USDC asset",
```

Comment:
This test is failing on the very last step. It is looking for the transaction on the wrong account (should be checking the USDC subAccount instead of the main XLM account). The transaction seems to be correct (example below) because it's successfully submitted to the ledger (can be viewed [here](https://stellar.expert/explorer/public/tx/212c2139c2485f241b13e5cdbceb48a9ee8a6383fae56d7d96f2b181fca6e7d4)).

There is a [`inferAccounts()`](https://github.com/LedgerHQ/ledger-live/pull/210/files#diff-13510fdeeb9e58cc283dbbbf65ec3f706a383a0c1757bdf8d8c06a752c8a120bR43) function in `cli-transaction.ts` file but it doesn't seem to be running (wrongly showing XLM in the transaction as well). Any idea what could be the issue here? :pray: 

![image](https://user-images.githubusercontent.com/7346473/170779512-1aab985d-7825-442d-8528-99e5b6b58c2e.png)
_Output of the bot_

![image](https://user-images.githubusercontent.com/7346473/170779225-3d0b8742-17f4-4b1c-9087-0274f47bc5ca.png)
_Transaction passed to the bot_